aiida/cmdline/params/options/__init__.py

```diff
@@ -315,10 +315,10 @@ def decorator(command):
 
 BROKER_VIRTUAL_HOST = OverridableOption(
     '--broker-virtual-host',
-    type=types.HostnameType(),
+    type=click.types.StringParamType(),
```

Comment:
Just to understand: Didn't the HOSTNAME_REGEX already exclude slashes?
https://github.com/sphuber/aiida-core/blob/dbbea0e3582894f8ef88e0f534db7560a1d8994d/aiida/cmdline/params/types/strings.py#L62
Why not keep the HostnameType?

Reply:
> Just to understand: Didn't the HOSTNAME_REGEX already exclude leading slashes?

Yes, which is exactly the problem. A virtual host with one or more slashes is perfectly valid. See also my commit message:

> On top of that, slashes would fail the validation outright, even though these are common in virtual hosts.

In the final paragraph of the commit message I explain why we resort to a type with as little restrictions as possible, namely a simple string type. I checked the RabbitMQ docs for virtual host naming rules but couldn't find it, so I asked on the mailing list and experimented: https://groups.google.com/u/1/g/rabbitmq-users/c/ZT56iwGbxy0   The last response is what I found by experimenting, almost anything goes. So I think it is best not to restrict anything in our interface.

Reply:
> Yes, which is exactly the problem. A virtual host with one or more slashes is perfectly valid. 

Well, I was going by the help string, in which you ask people *not* to add the leading slash.

But I see now that these virtual hosts can contain almost any string anywhere, thanks for the info!